chrome/content/zotero/preferences/preferences_sync.js

```diff
@@ -24,19 +24,125 @@
 */
 
 "use strict";
+Components.utils.import("resource://gre/modules/Services.jsm");
 
 Zotero_Preferences.Sync = {
-	init: function () {
+	init: Zotero.Promise.coroutine(function* () {
 		this.updateStorageSettings(null, null, true);
-		
-		document.getElementById('sync-api-key').value = Zotero.Sync.Data.Local.getAPIKey();
-		
+
+		var username = Zotero.Prefs.get('sync.server.username');
+		var apiKey = Zotero.Sync.Data.Local.getAPIKey();
+		if (apiKey) {
+			this.displayFields(username);
+			try {
+				var keyInfo = yield Zotero.Sync.Runner.checkAccess(Zotero.Sync.Runner.getAPIClient({apiKey}));
+				Zotero.Prefs.set('sync.server.username', keyInfo.value);
+				this.displayFields(keyInfo.username);
+			} catch (e) {
+				// API key wrong/invalid
```

Comment:
This needs to check `if (e instanceof Zotero.HTTP.UnexpectedStatusException)` and not unlink if so. Otherwise a temporary server error would unlink anyone who opened the pane. (That'd be a 5xx, but any 4xx that made it through (i.e., not 404, which is caught) should probably similarly be ignored.)